File: home/dot_local/share/lazyvim/lib/platforms/macos.mjs
export {
  configureNodeHost,
  configureUnixRuntimeEnvironment as configureRuntimeEnvironment,
  managedNeovimExecutable,
  managedNodeExecutable,
  managedToolExecutable,
  neovimDataDirectory,
  verifyNodeHost,
} from "./unix.mjs";

export const platformName = "darwin";

export function configureFonts() {}

export function verifyFonts() {
  const fontDirectory = path.join(
    targetHome,
    "Library",
    "Fonts",
    "JetBrainsMonoNerdFont",
  );
  const fontFiles = fs
    .readdirSync(fontDirectory)
    .filter((name) => name.endsWith(".ttf"));
  if (fontFiles.length === 0)
    throw new Error("No JetBrainsMono Nerd Font files installed");
  const catalog = captureCommandOutput("system_profiler", [
    "SPFontsDataType",
    "-json",
  ]);
  if (!catalog.toLowerCase().includes("jetbrainsmono")) {
    throw new Error("macOS font catalog cannot see JetBrainsMono Nerd Font");
  }
}
import fs from "node:fs";
import path from "node:path";
import { captureCommandOutput } from "../commands.mjs";
import { targetHome } from "../paths.mjs";

File: home/dot_local/share/lazyvim/lua/setup/platforms/macos.lua
local commands = require("setup.commands")
local paths = require("setup.paths")
return require("setup.platforms.unix").new({
	name = "darwin",
	configure_fonts = function() end,
	verify_fonts = function()
		local directory = paths.join(paths.home, "Library", "Fonts", "JetBrainsMonoNerdFont")
		local found = false
		for name in vim.fs.dir(directory) do
			if name:sub(-4) == ".ttf" then
				found = true
				break
			end
		end
		assert(found, "No JetBrainsMono Nerd Font files installed")
		local catalog = commands.capture("system_profiler", { "SPFontsDataType", "-json" }):lower()
		assert(catalog:find("jetbrainsmono", 1, true), "macOS font catalog cannot see JetBrainsMono Nerd Font")
	end,
})
